tests/my-game/example/struct-of-structs.ts

```diff
@@ -77,12 +77,12 @@ constructor(
 
 pack(builder:flatbuffers.Builder): flatbuffers.Offset {
   return StructOfStructs.createStructOfStructs(builder,
-    (this.a === null ? 0 : this.a.id!),
-    (this.a === null ? 0 : this.a.distance!),
-    (this.b === null ? 0 : this.b.a!),
-    (this.b === null ? 0 : this.b.b!),
-    (this.c === null ? 0 : this.c.id!),
-    (this.c === null ? 0 : this.c.distance!)
+    (this.a === null ? 0 : this.a?.id!),
```

Comment:
This seems quite odd - you first allow `a` to be undefined (?), but then you assert the value is present (!). Looks contradictory. The same pattern repeats for other fields in this PR.

Reply:
Of cause this looks a bit strange. We could simplify this line from `(this.a === null ? 0 : this.a?.id!)` to `(this.a?.id!)`. It is equivalent and should compile.

Reply:
This would still be wrong - the result of the `?.` operator can be undefined, so you can't later assert it's not by using `!`.

Reply:
What would you suggest?
Would this solution be ok? :
- struct of struct: `(this.a === null ? 0 : this.a.id!)` 
- struct of struct of struct: `(this.a?.b === null ? 0 : this.a.b.id!)` <- i think this will not compile

Reply:
Why do you need `?.` in the first place? There is already a null check present.

Reply:
Ok, I see what the problem might be for nested structs, but the proposed code doesn't solve it. For example, `this.a?.a === null` will never be true if `this.a` is null, thus defeating the check. Instead, something like this should work: `this.a?.a ?? 0`.

Reply:
You are right. The solution is:

- struct of struct: `this.a?.id && <nullvalue>`
- struct of struct of struct: `this.a?.b?.id && <nullvalue>`

<nullvalue> is `false` for type bool and `0` for all other simple data types.
I will try it out. If it works i will alter this pull request. I will also write a test case for this.

Reply:
No, this is not what I proposed - using `&&` instead of `??` will not work.

Reply:
Of cause - a typo at my side. Thanks